bulk_sync_into_benchling/bulk_sync_into_benchling.py

```diff
@@ -0,0 +1,205 @@
+    return rv.json()
+
+
+def api_post(domain, api_key, path, body):
+    url = "https://{}/api/v2/{}".format(domain, path)
+    rv = requests.post(url, json=body, auth=(api_key, ""))
+    if rv.status_code >= 400:
+        raise BadRequestException(
+            "Server returned status {}. Response:\n{}".format(
+                rv.status_code, json.dumps(rv.json())
+            ),
+            rv,
+        )
+    return rv.json()
+
+
+def wait_on_task_response(domain, api_key, task_resource):
+    task_id = task_resource["taskId"]
+    while True:
+        task_response = api_get(domain, api_key, "tasks/{}".format(task_id))
+        status = task_response["status"]
+        if status == "RUNNING":
+            time.sleep(10)
+        else:
+            if status == "FAILED":
+                return "FAILED", task_response["message"], task_response["errors"]
+            elif status == "SUCCEEDED":
+                return "SUCCEEDED", task_response["response"]
+
+
+@click.command()
+@click.option(
+    "--domain",
+    help="Domain name of your Benchling instance, e.g. example.benchling.com",
+    required=True,
+)
+@click.option("--api-key", help="Your API key", required=True)
+@click.option(
+    "--folder-id", help="ID of a folder to create the antibody in", required=True
+)
+@click.option("--registry-id", help="ID of the Benchling Registry", required=True)
+@click.option("--antibody-schema-id", help="ID of the Antibody schema", required=True)
+@click.option(
+    "--chain-schema-id",
+    help="ID of the Chain schema (Must be an AA-Sequence)",
+    required=True,
+)
+@click.argument("json_file_to_import", type=click.File("r"))
+def main(
+    domain,
+    api_key,
+    antibody_schema_id,
+    registry_id,
+    chain_schema_id,
+    folder_id,
+    json_file_to_import,
+):
+    antibodies_json = json.loads(json_file_to_import.read())
+    antibody_jsons = antibodies_json["antibodies"]
+
+    # Bulk create heavy chains into registry
+    task_resource = api_post(
+        domain,
+        api_key,
+        # https://docs.benchling.com/reference#create-custom-entity
+        "aa-sequences:bulk-create",
+        {
+            "aaSequences": [
+                {
+                    "aminoAcids": antibody_json["Heavy Chain"],
+                    "folderId": folder_id,
+                    "name": "Heavy Chain for {}".format(antibody_json["name"]),
+                    "schemaId": chain_schema_id,
+                    "registryId": registry_id,
+                    "namingStrategy": "NEW_IDS",
+                }
+                for antibody_json in antibody_jsons
+            ]
+        },
+    )
+    task_response = wait_on_task_response(domain, api_key, task_resource)
+    if task_response[0] == "FAILED":
+        print(
+            "Could not register at least one heavy chain. Error response from server:\n{}\n{}".format(
+                task_response[1],
+                task_response[2],
+            )
+        )
+        return
```

Comment:
Right now our bulk-create endpoints fail when the intended action violates a unique constraint. This causes an awkward experience which forces devs to constantly delete/unregister the entities they just create in a bad run. Should I add a check to get the aa-sequences if they already exist?

Reply:
They wouldn't actually delete/unregister entities, they would just remove the conflicting new ones from the bulk request. Unfortunately the bulk endpoint doesn't do a good job of returning error codes to make that easy right now. This is the same issue @loustalet was talking about yesterday. We probably want to fix these issues so we can write a clean example here.

Reply:
I guess I can put up this example and then update when the bulk-endpoint is cleaned up.